soft-interface: fix compilation on Kernel 5.4

cbricart opened the pull request:
Linux commit 895b5c9f206eb ("netfilter: drop bridge nf reset from nf_reset ")
renamed nf_reset to nf_reset_ct. Use the new function on >= 5.4 kernels.

fixes #17

T-X commented:
Hi @cbricart, could you use/add a define in compat.h instead? That's the usual way to handle compatibility in batman-adv.

Or ideally, backport this patch from upstream, so that we have the correct git history and titles from upstream as well:

https://git.open-mesh.org/batman-adv.git/commit/ab636a2bec851b103c638ec4cde059be0432528a

Thanks for looking into this issue!

cbricart commented:
a real backport does not seem to be possible, due to different file structure (i.e compat/linux/*.h vs. compat.h). (or at least my git-foo-powers are not sufficient ;) )
However, I tried to at least mimic backporting and moved the define to compat.h and updated soft-interface.c to follow mainline

T-X commented:
Looks better the diff, thanks!

For the commit message, ideally, you'd do something like:

```
$ wget --content-disposition "https://git.open-mesh.org/batman-adv.git/patch/ab636a2bec851b103c638ec4cde059be0432528a"
$ git am batman-adv.git-ab636a2bec851b103c638ec4cde059be0432528a.patch
- will complain, but you can continue adding your changes manually, then continue with:
$ git add compat.h soft-interface.c
- continue committing the patch:
$ git am --continue
$ git commit --amend
- add note about backport to the bottom, like:
  "[Christian Bricart: rebase to batadv-legacy]"
  (also compare with other backported patches in "$ git log")
- force push to update the PR, so that there is just one, clean commit, like:
$ git push <remote> --force
```

@cbricart  Do you want to try that or should I do that quickly?

cbricart commented:
thanks for the walkthrough @T-X 
but in the end for some reason, I rather had to revert my changes and then followed your walkthrough

so basically a squash-merge should do? - but I fail at doing that again ;)

I may also close this PR and create a new plain one

cbricart commented:
finally got it rebased

T-X commented:
Awesome, structure looks nice now!

One more thing, you now copied one "#endif" too many and it doesn't build. Once that's updated, I'll merge it.

cbricart commented:
oh.. my bad.. - fixed!
thank you

T-X commented:
Awesome, great work! And compiles fine for me now on a 5.6 kernel - merged.
